### DISCUSSION TITLE
Lab 3.01 Task 2

### DISCUSSION THREAD

**Discussion by claudiagiraldo** [rramosp/ai4eng.v1#250] [03 - GENERACION DE MODELOS]
Hola, tengo un problema realizando el task 2 de Lab 3.01

Este es mi código:

```
import itertools

def fit(X,y):
    def predict(X, t):
      
      for i in t:
          cond = (X[:, 0] < i[0]) & (X[:, 1] > i[1]) 
          y_hat = np.where(cond, 0, 1)
          if np.mean(y==y_hat) >= 0.9:
              best = [i[0], i[1]]
              return np.array(best)
          
    width, length = np.linspace(0, 1, 11), np.linspace(0, 1, 11)
    t = list((list(i) for i in itertools.product(width, length)))

    return predict(X, t)   
```

Me sale el siguiente error:
![image](https://user-images.githubusercontent.com/70002096/228640501-4ac4deab-61ab-4bc7-a8fe-20f30102a9c8.png)

En mi consola el código corre perfecto y devuelve un numpy array, tal como se pide.

Además, ejecutando varias veces esta celda superior me da el siguiente error:
![image](https://user-images.githubusercontent.com/70002096/228640927-78c3de02-1660-49a1-87de-3b549938d152.png)

Este es el código que utilicé para el task 1, que se ejecutó correctamente:
```
def predict(X, t):
  cond = (X[:, 0] < t[0]) & (X[:, 1] > t[1]) 
  y = np.where(cond, 0, 1) 

  return np.array(y)
```

Supongo que el error esta en el código del Task 1, además tengo entendido que este debe de reutilizarse en el Task 2 pero no encuentro la manera de hacerlo. 

Agradezco cualquier ayuda o consejo que puedan darme.

**Comment by imEag** [comment]
Hola Claudia, ejecutando tu código del task 2 me di cuenta de dos cosas:

1. no retorna el resultado correcto
2. En ocasiones retorna None

Entonces te recomiendo reutilizar el código del task 1 que si funciona correctamente. Para lograr esto es te propongo los siguientes pasos: 

1. Dentro de la función fit (task 2) definir la función predict del task 1 tal cual como es la original.
2. Crear la combinación de todos 
`
theta0 = np.linspace(0,1,11)
theta1 = np.linspace(0,1,11)
combinations = np.array(list(itertools.product(theta0, theta1)))
`
3. llamar la función predict múltiples veces, cada vez con una posición del array de combinaciones y guardar su resultado en un array
4. Rretornar el valor del array combinaciones que tenga el mejor desempeño

**Reply by claudiagiraldo** [reply]
Hola Emmanuel, tenías razón. 

El error en realidad estaba en el bucle que tomaba el primer valor de t que diera una exactitud mayor o igual a 0.9, en lugar del t que diera el mejor desempeño.

Gracias!
